### PR TITLE
Remove deprecated funcs for pmetric

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan <bogdandrutu@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 ### 🛑 Breaking changes 🛑
 
+- Remove deprecated `*DataPoint.Flags()` funcs. (#6017)
```
Comment (dmitryax) [inline CHANGELOG.md:7]:
```suggestion
- Replace deprecated `*DataPoint.Flags()` funcs with `*DataPoint.FlagsImmutable()`. (#6017)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,9 @@
...
+- Remove deprecated `MetricDataPointFlagsStruct` struct and `NewMetricDataPointFlagsStruct` func. (#6017)
+- Remove deprecated `MetricDataPointFlags` struct and `NewMetricDataPointFlags` func. (#6017)
```
Comment (dmitryax) [inline CHANGELOG.md:9]:
```suggestion
- Replace deprecated `MetricDataPointFlags` struct with `MetricDataPointFlagsImmutable`. (#6017)
- Remove deprecated `NewMetricDataPointFlags` func. (#6017)
```

**Comment by dmitryax** [comment]
Contrib needs to be updated :)

**Comment by bogdandrutu** [comment]
> Contrib needs to be updated :)

Nope, actually I should keep the type as deprecated. Contrib caught me on this.

File: CHANGELOG.md (hunk 2 above)
Comment (bogdandrutu) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 1 above)
Comment (bogdandrutu) [reply CHANGELOG.md]: Done.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**